### DISCUSSION TITLE
Persist terminal after a shell exits

### DISCUSSION THREAD

**Discussion by m-hilgendorf** [microsoft/vscode-discussions#57] [Extension Development QnA] [answered]
Is it possible to configure a `Terminal` instance created with `createTerminal` such that it remains visible after the shell program exits? 

For example, say you write a shell like this 

```c
#include <stdio.h>
#include <string.h>

int main () {
    char*  line = NULL;
    size_t len = 0;

    for(;;) {
        printf("> ");
        fflush(stdout);
        getline(&line, &len, stdin);
        if (strcmp(line, "exit\n") == 0)
            break;
        // ... shell impl ...
    }

    return 0;
}
```

The shell will exit if the user types "exit". If you compile and run a shell like this, then write an extension that opens the shell, when a user types "exit" the terminal panel immediately closes. 

I'd like the opened terminal to persist with a message like `shell exited with code 0` in this case, is there a way to do this with `createTerminal`?

The use case is allowing users to see the past terminal output in the event of an exit (usually due to a bug in the shell implementation). In the current behavior this isn't possible without extensive workarounds.

**Comment by Tyriar** [comment, marked as answer]
You can't do this currently. The feature request for this is tracked in https://github.com/microsoft/vscode/issues/70444, give it a 👍 to vote for it
